app/utils/navigators/PartnershipNavigator.scala

```diff
@@ -116,4 +124,29 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
       case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
-}
\ No newline at end of file
+
+  private def regionBasedNameNavigation(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(AreYouInUKId) match {
+      case Some(false) => NavigateTo.dontSave(routes.PartnershipRegisteredAddressController.onPageLoad())
+      case _ => NavigateTo.dontSave(routes.ConfirmPartnershipDetailsController.onPageLoad())
+    }
+  }
+
+  private def regionBasedNavigation(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(PartnershipRegisteredAddressId) flatMap { address =>
+      countryOptions.regions(address.country.getOrElse("")) match {
```

Comment:
Sticking with an Option would be neater and more readable than getOrElse(""). Match on Some(UK) etc

Reply:
Its not easy as we are using case object with optional field and  calling method needs string